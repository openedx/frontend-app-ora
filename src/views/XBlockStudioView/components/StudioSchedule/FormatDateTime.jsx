import React from 'react';
import PropTypes from 'prop-types';

import { useIntl } from '@edx/frontend-platform/i18n';
import messages from '../messages';

const FormatDateTime = ({ date }) => {
  const { formatMessage, formatDate } = useIntl();
  return (
    <span>{date ? formatDate(date, {
      year: 'numeric',
      month: 'long',
      day: 'numeric',
      hour: 'numeric',
      minute: 'numeric',
    }) : formatMessage(messages.notSetLabel)}
    </span>
  );
};

FormatDateTime.defaultProps = {
  date: null,
};
FormatDateTime.propTypes = {
  date: PropTypes.string,
};

export default FormatDateTime;
