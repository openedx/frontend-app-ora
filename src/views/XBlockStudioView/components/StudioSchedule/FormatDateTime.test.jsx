import { shallow } from '@edx/react-unit-test-utils';

import FormatDateTime from './FormatDateTime';

describe('<FormatDateTime />', () => {
  it('should render', () => {
    const wrapper = shallow(<FormatDateTime />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });

  it('should render with date', () => {
    const wrapper = shallow(<FormatDateTime date="2020-01-01T00:00:00Z" />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });
});
