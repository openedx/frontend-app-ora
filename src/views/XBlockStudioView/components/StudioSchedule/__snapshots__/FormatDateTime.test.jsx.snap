// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormatDateTime /> should render 1`] = `
<span>
  Not set
</span>
`;

exports[`<FormatDateTime /> should render with date 1`] = `
<span>
  2020-01-01T00:00:00.000Z
</span>
`;
